JustSaying.Extensions.DependencyInjection.Microsoft/IServiceCollectionExtensions.cs

```diff
@@ -124,8 +124,11 @@ public static IServiceCollection AddJustSaying(this IServiceCollection services,
             services.TryAddSingleton<IAwsClientFactoryProxy>((p) => new AwsClientFactoryProxy(p.GetRequiredService<IAwsClientFactory>));
             services.TryAddSingleton<IMessagingConfig, MessagingConfig>();
             services.TryAddSingleton<IMessageMonitor, NullOpMessageMonitor>();
-            services.TryAddSingleton<IMessageContextAccessor, MessageContextAccessor>();
-            services.TryAddSingleton<IMessageContextReader, MessageContextAccessor>();
+
+            var messageContextAccessor = new MessageContextAccessor();
```

Comment:
This isn't lazy, so IMHO the nicer way is to do it like this:

```csharp
services.TryAddSingleton<MessageContextAccessor>();
services.TryAddSingleton<IMessageContextAccessor>(serviceProvider => serviceProvider.GetRequiredService<MessageContextAccessor>());
services.TryAddSingleton<IMessageContextReader>(serviceProvider => serviceProvider.GetRequiredService<MessageContextAccessor>());
```

Example:

https://github.com/aspnet/Extensions/blob/557995ec322f1175d6d8a72a41713eec2d194871/src/HttpClientFactory/Http/src/DependencyInjection/HttpClientFactoryServiceCollectionExtensions.cs#L36-L38